File: info-status/StanfordPoS.java
package coref;

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileWriter;
import java.io.IOException;
import java.util.*;

import edu.stanford.nlp.ling.CoreAnnotations.*;
import edu.stanford.nlp.ling.CoreLabel;
import edu.stanford.nlp.pipeline.Annotation;
import edu.stanford.nlp.pipeline.StanfordCoreNLP;
import edu.stanford.nlp.util.CoreMap;

/**
 * Handles getting parts of speech from Stanford CoreNLP and exports into a
 * text file with the words and their parts of speech
 * @author Kara Schechtman kws2121
 * @version 01.09.2018
 */
public class StanfordPoS {
	private static int TOTAL_SESSION_NUMS = 12;
	private static String ORDER_TYPE = "INTERPOLATED";
	/**
	 * Main method to read in file and output to stdout the coreference chains
	 * @param args
	 * @throws Exception
	 */
	public static void main(String[] args) throws Exception {
		parsePoS();
	}

	/**
	 * Run PoS tagging on files 01-12.txt and save results in format ##words.txt
	 * @throws IOException
	 */
	public static void parsePoS() throws IOException {
		for (int i = 1; i <= TOTAL_SESSION_NUMS;i++) {
			String formatted = String.format("%02d", i);
			stanfordTag(formatted + "_" + ORDER_TYPE + ".txt");
		}
	}

	/**
	 * Helper to read in the text of filename, parse using StanfordNLP parser, and save all the PoS tags to a file
	 * @param filename the filename to read
	 */
	private static void stanfordTag(String filename) throws IOException {
		Annotation document = null;
		try {
			document = new Annotation(readFile(filename));
		} catch (FileNotFoundException e) {
			// catch block
			e.printStackTrace();
		}
		Properties props = new Properties();
		props.setProperty("annotators", "tokenize,ssplit,pos,lemma,ner,parse,mention,coref");
		StanfordCoreNLP pipeline = new StanfordCoreNLP(props);
		pipeline.annotate(document);
		List<CoreMap> sentences = document.get(SentencesAnnotation.class);
		String num = filename.substring(0, filename.lastIndexOf("."));
		String save = num + "words.txt";

		BufferedWriter output = null;
		try {
		    File file = new File(save);
		    output = new BufferedWriter(new FileWriter(file));
		    for(CoreMap sentence: sentences) {
				  for (CoreLabel token: sentence.get(TokensAnnotation.class)) {
				    // text of the word
				    String word = token.get(TextAnnotation.class);
				    // POS tag of the word
				    String pos = token.get(PartOfSpeechAnnotation.class);
				    String postag = word + " " + pos + "\n";
				    System.out.println(postag);
				    output.write(postag);
				  }
			}
			output.close();
		} catch ( IOException e ) {
			e.printStackTrace();
		}
	}

	/**
	 * Helper to read in a file into a string
	 * @param filename the file to read
	 * @return the string of the filetext
	 * @throws FileNotFoundException
	 */
	private static String readFile(String filename) throws FileNotFoundException {
		Scanner scanner = new Scanner(new File(filename));
		String entireFileText = scanner.useDelimiter("\\A").next();
		scanner.close();
		return entireFileText;
	}
}

File: info-status/StanfordPoS.py
import json
import logging
import os
import pandas as pd
import string
from time import sleep
from stanfordcorenlp import StanfordCoreNLP
from OrganizedBigTable import OrganizedBigTable, OrderType, _turnsToText
from CombineSessionTables import combine
from example_config import config

CORE_NLP_PATH = config.get('core_nlp_path')
START_SESSION = config.get('start_session')
END_SESSION = config.get('end_session')
ORDER_TYPE = config.get('order_type')
MEMORY = config.get('java_memory') or '4g'

props = {'annotators': 'tokenize,ssplit,pos','outputFormat':'json'}

def parse_pos():
    with StanfordCoreNLP(CORE_NLP_PATH, memory=MEMORY, quiet=True) as client:
        for i in range(START_SESSION, END_SESSION+1):
            session_number = str(i).zfill(2)
            filename = "{}_{}.txt".format(session_number, ORDER_TYPE)
            stanford_tag(client, filename)
        sleep(5) # make sure all sessions complete before closing client

def stanford_tag(client, filename):
    document = open(filename)
    text = document.read()
    ann = client.annotate(text, properties=props)
    parsed = json.loads(ann)
    words_file_name = '{}words.txt'.format(os.path.splitext(filename)[0])
    words_file = open(words_file_name, 'w')
    for item in parsed['sentences']:
        for token in item['tokens']:
            word = token['word']
            pos = token['pos']
            words_file.write('{} {}\n'.format(word, pos))

    document.close()
    words_file.close()

if __name__ == '__main__':
    parse_pos()
